### PR TITLE
Ticket31820

### PR DESCRIPTION
**Pull request by nmathewson**

### CODE REVIEW

File: src/lib/crypt_ops/aes_openssl.c [hunk 1]
```diff
@@ -111,19 +61,22 @@ aes_new_cipher(const uint8_t *key, const uint8_t *iv, int key_bits)
   EVP_EncryptInit(cipher, c, key, iv);
   return (aes_cnt_cipher_t *) cipher;
 }
+/**
```
Comment (teor2345) [inline src/lib/crypt_ops/aes_openssl.c]: Nitpick: newline?
And for the other functions following this one.

File: src/lib/crypt_ops/crypto_openssl_mgt.c [hunk 2]
```diff
@@ -128,46 +117,11 @@ crypto_openssl_get_header_version_str(void)
...
-tor_set_openssl_thread_id(CRYPTO_THREADID *threadid)
-{
-  CRYPTO_THREADID_set_numeric(threadid, tor_get_thread_id());
-}
-#endif /* !defined(NEW_THREAD_API) */
-
 /** Helper: Construct mutexes, and set callbacks to help OpenSSL handle being
  * multithreaded. Returns 0. */
 static int
 setup_openssl_threading(void)
```
Comment (teor2345) [inline src/lib/crypt_ops/crypto_openssl_mgt.c]: Should we just delete this empty function?

File: src/lib/crypt_ops/compat_openssl.h [hunk 3]
```diff
@@ -21,37 +21,32 @@
...
-#ifndef OPENSSL_1_1_API
-#define OpenSSL_version(v) SSLeay_version(v)
-#define OpenSSL_version_num() SSLeay()
+#ifdef LIBRESSL_VERSION_NUMBER
 #define RAND_OpenSSL() RAND_SSLeay()
 #define STATE_IS_SW_SERVER_HELLO(st)       \
   (((st) == SSL3_ST_SW_SRVR_HELLO_A) ||    \
    ((st) == SSL3_ST_SW_SRVR_HELLO_B))
 #define OSSL_HANDSHAKE_STATE int
 #define CONST_IF_OPENSSL_1_1_API
```
Comment (teor2345) [inline src/lib/crypt_ops/compat_openssl.h]: Perhaps you could rename this to CONST_IF_OPENSSL ?

File: src/lib/tls/tortls_openssl.c [hunk 4]
```diff
@@ -313,15 +302,9 @@ tor_tls_init(void)
...
-    SSL_library_init();
-    SSL_load_error_strings();
-#endif /* defined(OPENSSL_1_1_API) */
 
-#if (SIZEOF_VOID_P >= 8 &&                              \
-     OPENSSL_VERSION_NUMBER >= OPENSSL_V_SERIES(1,0,1))
+#if (SIZEOF_VOID_P >= 8)
     long version = OpenSSL_version_num();
 
     /* LCOV_EXCL_START : we can't test these lines on the same machine */
```
Comment (teor2345) [inline src/lib/tls/tortls_openssl.c]: You missed a `OPENSSL_V_SERIES(1,0,1)` just below this line.

File: src/lib/crypt_ops/aes_openssl.c (hunk 1 above)
Comment (nmathewson) [reply src/lib/crypt_ops/aes_openssl.c]: Done in 3acadb6d890391d547252a282918ea1bab0e2078

File: src/lib/crypt_ops/crypto_openssl_mgt.c (hunk 2 above)
Comment (nmathewson) [reply src/lib/crypt_ops/crypto_openssl_mgt.c]: Done in 032479966875b474646e6234690ac9a97bd1771b

File: src/lib/crypt_ops/compat_openssl.h (hunk 3 above)
Comment (nmathewson) [reply src/lib/crypt_ops/compat_openssl.h]: Done in e5e206966ad3f7

File: src/lib/tls/tortls_openssl.c (hunk 4 above)
Comment (nmathewson) [reply src/lib/tls/tortls_openssl.c]: fixed as bd73669874ed9e

**Pull request closed without merging**